subsys/dm/dm.c

```diff
@@ -167,12 +180,11 @@ static mpsl_timeslot_signal_return_param_t *mpsl_timeslot_callback(
 			return p_ret_val;
 		}
 
-		nrf_dm_status = dm_configure();
+		dm_config_get(&timeslot_ctx.curr_req.dm_req, &dm_config);
+		nrf_dm_status = nrf_dm_configure(&dm_config);
 
 		if (nrf_dm_status == NRF_DM_STATUS_SUCCESS) {
-			uint32_t timeout = timeslot_ctx.curr_req.dm_req.role == DM_ROLE_INITIATOR ?
-					 INITIATOR_RANGING_WINDOW_US : REFLECTOR_RANGING_WINDOW_US;
-			nrf_dm_status = nrf_dm_proc_execute(timeout);
+			nrf_dm_status = nrf_dm_proc_execute(timeslot_ctx.curr_req.window_length);
 		}
```

Comment:
I realise this is unrelated to you current PR, but if the  `nrf_dm_configure` call does not return `NRF_DM_STATUS_SUCCESS` then some invalid parameters are likely being set in the `dm_config`. Maybe this should be logged somehow.

Reply:
Added log message, if the returned value is not `NRF_DM_STATUS_SUCCESS`